build-system/tasks/performance/measure-documents.js

```diff
@@ -70,6 +74,35 @@ const setupMeasurement = page =>
     });
   });
 
+/**
+ * Handles request interception
+ *
+ * @param {Request} interceptedRequest
+ * @param {Array<number>} endTimes
+ */
+async function handleRequests(interceptedRequest, endTimes) {
+  const interceptedUrl = interceptedRequest.url();
+  const matchArray = interceptedUrl.match(CDN_ANALYTICS_REGEXP);
+  if (matchArray) {
```

Comment:
Wondering if this is a good opportunity to setup an extensible pattern here, possibly some mapping of URLs to handlers since it seems likely there would be more cases?

Reply:
That's a good point! I know the ads team want to collect two requests delay. Not sure if they'll use this specific task. Maybe useful to expand the feature here. 
@powerivq Any comment? Thanks

Reply:
I am thinking of this: each team still implements its own handleRequest to respond the payload it desires, and we make another interceptor to record the timing of the request. Does it make sense? @zhouyx

Reply:
@kevinkimball good point!

Reply:
@powerivq I tested out multiple request handlers and it seems to work fine. Seems like a good route to go down in the future.

Reply:
Looked at the definition of `networkidle0`. I don't know if this would satisfy ads team's request. ActiveView requires the ad to be visible for at least 50% for 1 second. It's very possible that the task ends before activeView. @kevinkimball @micajuine-ho any workaround?

Reply:
I believe just adding a [setTimeout](https://github.com/puppeteer/puppeteer/issues/651) would do the trick, however I'm not sure if we want to always include it. Maybe we could add a flag (`--set-timeout=500`) that would set how long to wait for, if any. Thoughts @zhouyx ?

Reply:
A race between timeout, and the analytics request sounds good to me. 
My recommendation is to always include the timeout, but set it to a value that's long enough. People will go with the default setting most of the time, and a timeout will serve as a fallback and reduce flakiness.

BTW this is something I'm testing with, please ignore the console log part : )

```
let timeout_count = 0;
const promise = new Promise(resolve => {
    r = resolve;
    setTimeout(() => {
      timeout_count++;
      resolve();
   }, TIMEOUT);
});
  page.on('console', message => {
    if (String(message._text).includes('request delay')) {
      const delay = Number(message._text.replace('analytics request delay is  ', ''));
      all.push(delay);
      r();
    }
  });
.....
await promise; 
```

Reply:
@zhouyx Ok that sounds good to me

Reply:
Add @estherkim for the best way to customize timeout for different test cases.